Collections_Homework2.py

```diff
@@ -0,0 +1,21 @@
+from string import ascii_lowercase # we will use it to generate keys as letters
+from random import randint, choice # we will use it to generate random integer for dictionary values and list length; choice will be used to select random letter for keys
+
+rand_lst = [{choice(ascii_lowercase): randint(0,100) for i in range(len(ascii_lowercase))} for j in range(randint(2,10))] # list of random dicts
```

Comment:
Perfect!

Reply:
Tested your code on followind dataset:

rand_lst = [{'m': 84, 'f': 41, 'l': 34, 'c': 33, 'q': 58, 'h': 14, 'p': 25, 't': 85, 'x': 70, 'g': 3, 'r': 63, 'y': 93, 'd': 44, 'o': 79, 'n': 73},
 {'n': 86, 'l': 1, 'w': 68, 'h': 52, 'e': 91, 'x': 31, 'f': 58, 'b': 45, 'a': 45, 'q': 100, 'r': 39, 'p': 88, 'z': 27, 'j': 66},
 {'p': 35, 'd': 73, 'j': 75, 'v': 89, 'm': 81, 'l': 33, 'z': 60, 'h': 59, 'u': 82, 's': 96, 'i': 86, 'b': 68, 'w': 100, 'x': 38, 'a': 50, 'o': 67},
 {'p': 26, 'f': 74, 'o': 63, 'v': 4, 'e': 2, 'r': 39, 'k': 34, 'w': 95, 'g': 51, 'n': 47, 'h': 27, 'c': 29, 'z': 0, 'q': 33, 'a': 69, 't': 87}]

Final dictionary: 
{'m': 84, 'l': 34, 'c': 33, 'x': 70, 'r': 63, 'y': 93, 'o': 79, 'n_1': 86, 'e': 91, 'q_1': 100, 'p_1': 88, 'd_2': 73, 'j_2': 75, 'v': 89, 'z_2': 60, 'h_2': 59, 'u': 82, 's': 96, 'i': 86, 'b_2': 68, 'w_2': 100, 'f_3': 74, 'k': 34, 'g_3': 51, 'a_3': 69, 't_3': 87}

**Expected Result: 'm1': 84**
Actual Result: 'm': 84

**Expected Result:** (depends on expected task logic)'g_2': 51 (2nd occurrence of "g") or 'g_4': 51 (4th dict starting from 1)
Actual Result: 'g_3': 51

Reply:
If you are interested in my idea:
```
from random import randint, choice
from string import ascii_lowercase


rand_list = [{choice(ascii_lowercase): randint(0, 100) for i in range(len(ascii_lowercase))} for j in range(randint(2, 10))]
print(rand_list)
final_dict, tmp_dict = {},  {}

#Transform from list of dicts into dict of lists.
for dictionary in rand_list:
    for k, v in dictionary.items():
        tmp_dict.setdefault(k, []).append(v)

#Now choose only the biggest one
for k, v in tmp_dict.items():
    if len(v) > 1:
        final_dict[k+"_"+str(v.index(max(v))+1)] = max(v)
    else:
        final_dict[k] = v[0]
print(final_dict)
```

Reply:
Thank you @ValentynTroian !

Your implementation is definitely much more elegant and readable, i need to learn more about setdefault)

I think I got the requirements wrong, my understanding was that:
1) If the value for the key encountered for the first time is max, leave key as is (that's why i have m and not m_1)
2) Occurrences = list indexes, so g_3 in my case is 4th dictionary

Do you mind if I update my code and commit the fix version?

Reply:
I completely understand that it's slightly hard to get requirements :) 

My script logic was based on Denys's example:

![11](https://user-images.githubusercontent.com/34323554/106747370-7e63dc80-662c-11eb-9714-dbdfbb96b944.PNG)

Yes, of course, you can update your code if you want. The only thing that I'm not sure that I can update your mark. But anyway 4.5 is a great result :)

Reply:
I've asked about the mark update process - I can do that before the start of next (4th module)

Reply:
Thank you @ValentynTroian !

I re-read the requirements - I totally overlooked that example. I saw "rename with dict number" and decided that's element index)

Anyway, you don't have to worry about the mark, I agree 4.5 is high enough)

I still committed the fixed version - would be great if you could review it whenever you have time). There are 2 versions in the latest commit:

1st version (non-commented): key is renamed based on sequence of occurrences of this key (this version should match your results, if I did everything correctly)
2nd version (commented): key is renamed based on dictionary number, but this time dict number is actual sequence (1 = 1st etc).

Also, the bug with not sequencing keys with max value on first occurrence (m: 84 instead of m_1: 84 in your earlier comment) is fixed.

Thank you!